src/stores/game.js

```diff
@@ -366,10 +368,13 @@ export const useGameStore = defineStore('game', {
       }
     },
     makeSocketRequest(slug, data, method = 'get') {
-      const url =
-        import.meta.env.VITE_USE_GAMESTATE_API === 'true'
-          ? `/api/gamestate/${this.id}/move/${slug}`
-          : `/api/game/${slug}`;
+      const usingGameStateAPI = import.meta.env.VITE_USE_GAMESTATE_API === 'true';
+      let url = `/api/game/${slug}`;
+  
+      if (usingGameStateAPI && !nonGameStateAPIMoves.has(slug)) {
```

Comment:
This block of logic feels like it should be pulled out in to a separate utility and tested to validate the proper call goes out for the specific APIs when the env is set.